### PR TITLE
refactor:create interval:use Json instead of Toml, use interactive mode

### PR DESCRIPTION
**Pull request by difince**
Refactor create interval command to work with Json format instead of Toml.
There are two ways of creating an Interval:
- by using a file that could contain one or more Interval configurations
- by using all other flags that auto generate Json Interval with values passed in with flags like --name, --frequency, --start etc.
If user provides -i flag, the default editor will be opened with the autogenerated Interval loaded ready for additional customizations.

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/interval/add/add.go [hunk 1]
```diff
@@ -15,60 +15,181 @@
 package add
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"github.com/edgexfoundry-holding/edgex-cli/pkg/editor"
```
Comment (mkbhanda) [inline cmd/interval/add/add.go:22]: drop the -holding

File: cmd/interval/add/add.go [hunk 2]
```diff
@@ -15,60 +15,181 @@
...
 func NewCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "add",
 		Short: "Add interval",
-		Long:  `Create the intervals described in the given TOML files.`,
-		Run:   addIntervalHandler,
+		Long:  `Create interval(s) described in the given JSON file or use the interactive mode with additional flags.`,
+		RunE:  newIntervalHandler,
 	}
+	cmd.Flags().BoolVarP(&interactiveMode, editor.InteractiveModeLabel, "i", false, "Open a default " +
```
Comment (mkbhanda) [inline cmd/interval/add/add.go:65]: s/a default/the default

File: cmd/interval/add/add.go [hunk 3]
```diff
@@ -15,60 +15,181 @@
...
+		"editor to customize the Interval information")
+	cmd.Flags().StringVarP(&name, "name", "n", "", "Interval name")
+	cmd.Flags().StringVarP(&start, "start", "s", "", "Interval Start time in format YYYYMMDD'T'HHmmss")
+	cmd.Flags().StringVarP(&end, "end", "e", "", "Interval End time in format YYYYMMDD'T'HHmmss")
+	cmd.Flags().StringVarP(&cron, "cron", "c", "", "Styled regular expression indicating " +
```
Comment (mkbhanda) [inline cmd/interval/add/add.go:70]: Might be have a conflict if cron is specified in conjunction with runOnce and or frequency?

File: cmd/interval/interval.go [hunk 4]
```diff
@@ -15,10 +15,10 @@
 package interval
 
 import (
-	addinterval "github.com/edgexfoundry-holding/edgex-cli/cmd/interval/add"
-	listinterval "github.com/edgexfoundry-holding/edgex-cli/cmd/interval/list"
-	rminterval "github.com/edgexfoundry-holding/edgex-cli/cmd/interval/rm"
-	updateinterval "github.com/edgexfoundry-holding/edgex-cli/cmd/interval/update"
+	"github.com/edgexfoundry-holding/edgex-cli/cmd/interval/add"
```
Comment (mkbhanda) [inline cmd/interval/interval.go:18]: may be just an artifact of the PR order .. but holding is mentioned here.

**Review by mkbhanda — COMMENTED** [review]
Diana do we need to take care of conflicting/multiple flags being presented cron/runonce/frequency?
Rest looks great/

**Review by mkbhanda — APPROVED** [review]
Diana shall merge this in but curious if we need to check in CLI whether there arises any conflict if cron, frequency, once are  provided together

File: cmd/interval/interval.go (hunk 4 above)
Comment (difince) [reply cmd/interval/interval.go]: yes, all these imports that contains "holding" are removed by PR https://github.com/edgexfoundry/edgex-cli/pull/279

File: cmd/interval/add/add.go (hunk 3 above)
Comment (difince) [reply cmd/interval/add/add.go:73]: Helpers messages I took from go-mod-core-contracts/API descriptions. The validation process in entirely done by  go-mod-core-contracts Interval.validate method.

File: cmd/interval/add/add.go (hunk 2 above)
Comment (difince) [reply cmd/interval/add/add.go:68]: I will include this fix within my next PR

**Comment by difince** [comment]
I took the helper message from go-mod-core-contracts/API descriptions. The input validation process is entirely done by go-mod-core-contracts Interval.validate method.
I already tested the creation of Interval that contains all three params - and it was successful. I guess that there is a priority ... the implementation needs to be checked

File: cmd/interval/interval.go (hunk 4 above)
Comment (difince) [reply cmd/interval/interval.go]: It looks that the conflicts between PR #279 and this one haven't need resolved correctly. So now the CLI cannot compile. I will open new issue, and will remove a few places where import "holding " is still present

**Pull request merged**